plugins/huggingface/tests/test_huggingface_api.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+from unittest.mock import ANY, patch
+
+from modelgauge.auth.huggingface_inference_token import HuggingFaceInferenceToken
+from modelgauge.prompt import SUTOptions, TextPrompt
+from modelgauge.sut import SUTCompletion, SUTResponse
+from modelgauge.suts.huggingface_api import (
+    HuggingFaceChatParams,
+    HuggingFaceChatRequest,
+    HuggingFaceResponse,
+    HuggingFaceSUT,
+)
+
+
+@pytest.fixture
+def fake_sut():
+    return HuggingFaceSUT("fake_uid", "https://fake_url.com", HuggingFaceInferenceToken("fake_token"))
+
+
+@pytest.fixture
+def prompt():
+    return TextPrompt(
+        text="some text prompt",
+        options=SUTOptions(max_tokens=5, temperature=1.0, random="random"),
+    )
+
+
+@pytest.fixture
+def sut_request():
+    return HuggingFaceChatRequest(
+        inputs="some text prompt", parameters=HuggingFaceChatParams(max_new_tokens=5, temperature=1.0)
+    )
+
+
+def test_huggingface_api_translate_text_prompt_request(fake_sut, prompt, sut_request):
+    request = fake_sut.translate_text_prompt(prompt)
+
+    assert isinstance(request, HuggingFaceChatRequest)
+    assert request == sut_request
+
+
+def mocked_requests_post():
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return [self.json_data]
+
+    return MockResponse({"generated_text": "response"}, 200)
+
+
+@patch("requests.post", side_effect=mocked_requests_post)
+def test_huggingface_api_evaluate_receives_correct_args(mock_post, fake_sut, sut_request):
+    fake_sut.evaluate(sut_request)
+
+    mock_post.assert_called_with(
+        "https://fake_url.com",
+        headers=ANY,
+        json={"inputs": "some text prompt", "parameters": {"max_new_tokens": 5, "temperature": 1.0}},
+    )
+
+
+@patch("requests.post", side_effect=mocked_requests_post)
+def test_huggingface_api_evaluate_dumps_result(mock_post, fake_sut, sut_request):
+    output = fake_sut.evaluate(sut_request)
+
+    assert output == HuggingFaceResponse(generated_text="response")
```

Comment:
Same concern here; to know what the test is testing you have to look elsewhere, and then it's not clear which are the important bits. It's best to think of tests as documenting your intent for programmers who come along later, but in a way that computers can verify still hold.